### ISSUE TITLE
Bluetooth devices have disappeared after 2.0.1

### ISSUE THREAD

**Issue by darkkatarsis** [AlexxIT/XiaomiGateway3#630]
Info: Gateway: v1.5.0_0026
BT device: BGX-5/X1-3001

After the update to version v2, my bluetooth device is no longer visible - it does not return any entities. I tried to restart the gateway and also removing / adding integrations.

Edit:
Downgrade to 1.6.5 fixes the problem

**Comment by AlexxIT** [comment]
@mikeinblue lock is known problem. Will be fixed.

@darkkatarsis can you check did of your device in logs. Is it `blt...` or it is a number

**Comment by dscao** [comment]
> 2022-02-09 18:09:36  DEBUG    base          192.168.8.20 [MQTT] log/ble b'{"did":"1020275136","eid":11,"edata":"a200000000de920362","pdid":1433,"seq":127}'
> 2022-02-09 18:09:36  DEBUG    base          192.168.8.20 [BASE] Unregistered BLE device {'did': '1020275136', 'eid': 11, 'edata': 'a200000000de920362', 'pdid': 1433, 'seq': 127}
> 2022-02-09 18:09:37  DEBUG    base          192.168.8.20 [MQTT] log/miio b'\x1b[0;32m2022:02:09:18:09:36.087 [D] miio_client_func: ot_agent_recv_handler_one(): fd:9, msg:{"method":"_async.ble_event","params":{"dev":{"did":"1020275136","mac":"E5:0B:9A:2D:2A:9F","pdid":1433},"evt":[{"eid":11,"edata":"a200000000de920362"}],"frmCnt":127,"gwts":1644401376},"id":1989} length:194 bytes\x1b[0m'
> 2022-02-09 18:09:37  ERROR    base          192.168.8.20 | Processing MQTT: log/miio b'\x1b[0;32m2022:02:09:18:09:36.087 [D] miio_client_func: ot_agent_recv_handler_one(): fd:9, msg:{"method":"_async.ble_event","params":{"dev":{"did":"1020275136","mac":"E5:0B:9A:2D:2A:9F","pdid":1433},"evt":[{"eid":11,"edata":"a200000000de920362"}],"frmCnt":127,"gwts":1644401376},"id":1989} length:194 bytes\x1b[0m'
> 2022-02-09 18:09:38  DEBUG    base          192.168.8.20 [MQTT] log/ble b'{"did":"1020275136","eid":11,"edata":"a300000000df920362","pdid":1433,"seq":128}'
> 2022-02-09 18:09:38  DEBUG    base          192.168.8.20 [BASE] Unregistered BLE device {'did': '1020275136', 'eid': 11, 'edata': 'a300000000df920362', 'pdid': 1433, 'seq': 128}
> 2022-02-09 18:09:38  DEBUG    base          192.168.8.20 [MQTT] log/miio b'\x1b[0;32m2022:02:09:18:09:38.617 [D] miio_client_func: ot_agent_recv_handler_one(): fd:9, msg:{"method":"_async.ble_event","params":{"dev":{"did":"1020275136","mac":"E5:0B:9A:2D:2A:9F","pdid":1433},"evt":[{"eid":11,"edata":"a300000000df920362"}],"frmCnt":128,"gwts":1644401377},"id":1990} length:194 bytes\x1b[0m'
> 2022-02-09 18:09:38  ERROR    base          192.168.8.20 | Processing MQTT: log/miio b'\x1b[0;32m2022:02:09:18:09:38.617 [D] miio_client_func: ot_agent_recv_handler_one(): fd:9, msg:{"method":"_async.ble_event","params":{"dev":{"did":"1020275136","mac":"E5:0B:9A:2D:2A:9F","pdid":1433},"evt":[{"eid":11,"edata":"a300000000df920362"}],"frmCnt":128,"gwts":1644401377},"id":1990} length:194 bytes\x1b[0m'
> 2022-02-09 18:09:39  DEBUG    base          192.168.8.20 [MQTT] log/ble b'{"did":"1020275136","eid":4106,"edata":"64e1920362","pdid":1433,"seq":129}'
> 2022-02-09 18:09:39  DEBUG    base          192.168.8.20 [BASE] Unregistered BLE device {'did': '1020275136', 'eid': 4106, 'edata': '64e1920362', 'pdid': 1433, 'seq': 129}
> 2022-02-09 18:09:41  DEBUG    base          192.168.8.20 [MQTT] log/ble b'{"did":"blt.3.14t8gpcjge800","eid":4105,"edata":"2900","pdid":152,"seq":100}'
> 2022-02-09 18:09:41  DEBUG    base          192.168.8.20 [BASE] Unregistered BLE device {'did': 'blt.3.14t8gpcjge800', 'eid': 4105, 'edata': '2900', 'pdid': 152, 'seq': 100}
> 2022-02-09 18:09:45  DEBUG    base          192.168.8.20 [MQTT] log/ble b'{"did":"blt.3.14bvdc8qkeg00","eid":4119,"edata":"58020000","pdid":2038,"seq":113}'
> 2022-02-09 18:09:45  DEBUG    base          192.168.8.20 [BASE] Unregistered BLE device {'did': 'blt.3.14bvdc8qkeg00', 'eid': 4119, 'edata': '58020000', 'pdid': 2038, 'seq': 113}
> 2022-02-09 18:09:46  DEBUG    base          192.168.8.20 [MQTT] log/miio b'\x1b[0;32m2022:02:09:18:09:45.267 [D] miio_client_func: ot_agent_recv_handler_one(): fd:9, msg:{"method":"_async.ble_event","params":{"dev":{"did":"blt.3.14bvdc8qkeg00","mac":"50:EC:50:CE:93:07","pdid":2038},"evt":[{"eid":4119,"edata":"58020000"}],"frmCnt":113,"gwts":1644401385},"id":1992} length:195 bytes\x1b[0m'
> 2022-02-09 18:09:46  DEBUG    base          192.168.8.20 [BASE] 50ec50ce9307 (2038) setup ble MJYD02YL-A: battery, light, motion, idle_time
> 2022-02-09 18:09:46  DEBUG    base          192.168.8.20 [BLEE] 50ec50ce9307 (2038) recv {'idle_time': 600}

Unregistered BLE device {'did': '1020275136', 'eid': 11, 'edata': 'a200000000de920362', 'pdid': 1433, 'seq': 127}

pdid: 1433 is exist in device.py, but show "Unregistered BLE device."

**Comment by darkkatarsis** [comment]
@AlexxIT 
Logs:
```
2022-02-10 10:09:29  DEBUG    main          SysInfo: {'installation_type': 'Home Assistant OS', 'version': '2022.2.5', 'dev': False, 'hassio': True, 'virtualenv': False, 'python_version': '3.9.7', 'docker': True, 'arch': 'x86_64', 'timezone': 'Europe/Warsaw', 'os_name': 'Linux', 'os_version': '5.10.91', 'user': 'root', 'supervisor': '2022.01.1', 'host_os': 'Home Assistant OS 7.2', 'docker_version': '20.10.9', 'chassis': 'vm'}
2022-02-10 10:09:29  DEBUG    base          192.168.0.9 [BASE] Start main loop
2022-02-10 10:09:29  DEBUG    base          192.168.0.9 [BASE] Prepare gateway gw3 with firmware 1.5.0_0026
2022-02-10 10:09:29  DEBUG    base          192.168.0.9 [BASE] 54ef4424f543 (lumi.gateway.mgl03) setup gateway ZNDMWG03LM ZNDMWG02LM: alarm, command, data, cloud_link, led, gateway
2022-02-10 10:09:29  DEBUG    base          192.168.0.9 [BASE] Patch Bluetooth
2022-02-10 10:09:29  DEBUG    base          192.168.0.9 [BASE] Applied 0 patches to daemons
2022-02-10 10:09:29  DEBUG    base          192.168.0.9 [BASE] MQTT connected
2022-02-10 10:09:29  DEBUG    base          192.168.0.9 [MQTT] ble/50325f17f418 b'{"opening":0,"battery":88}'
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [BASE] Stop all tasks
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [BASE] Start main loop
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [BASE] Prepare gateway gw3 with firmware 1.5.0_0026
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [BASE] 54ef4424f543 (lumi.gateway.mgl03) setup gateway ZNDMWG03LM ZNDMWG02LM: alarm, command, data, cloud_link, led, gateway
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [BASE] Patch Bluetooth
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [BASE] Applied 0 patches to daemons
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [BASE] MQTT connected
2022-02-10 10:09:33  DEBUG    base          192.168.0.9 [MQTT] ble/50325f17f418 b'{"opening":0,"battery":88}'
2022-02-10 10:09:38  DEBUG    base          192.168.0.9 [MQTT] log/ble b'{"did":"1022910617","eid":4111,"edata":"01","pdid":2480,"seq":225}'
2022-02-10 10:09:38  DEBUG    base          192.168.0.9 [BASE] Unregistered BLE device {'did': '1022910617', 'eid': 4111, 'edata': '01', 'pdid': 2480, 'seq': 225}
2022-02-10 10:09:41  DEBUG    base          192.168.0.9 [MQTT] log/ble b'{"did":"1022910617","eid":4106,"edata":"58","pdid":2480,"seq":226}'
2022-02-10 10:09:41  DEBUG    base          192.168.0.9 [BASE] Unregistered BLE device {'did': '1022910617', 'eid': 4106, 'edata': '58', 'pdid': 2480, 'seq': 226}
2022-02-10 10:10:33  DEBUG    base          192.168.0.9 [MQTT] gw/BC33ACFFFE7E8BCC/heartbeat b'{"networkUp":true,"networkPanId":"0x80F5","radioTxPower":7,"radioChannel":11}'
2022-02-10 10:10:44  DEBUG    base          192.168.0.9 [MQTT] log/ble b'{"did":"1022910617","eid":4111,"edata":"01","pdid":2480,"seq":228}'
2022-02-10 10:10:44  DEBUG    base          192.168.0.9 [BASE] Unregistered BLE device {'did': '1022910617', 'eid': 4111, 'edata': '01', 'pdid': 2480, 'seq': 228}
2022-02-10 10:10:47  DEBUG    base          192.168.0.9 [MQTT] log/ble b'{"did":"1022910617","eid":4106,"edata":"58","pdid":2480,"seq":229}'
2022-02-10 10:10:47  DEBUG    base          192.168.0.9 [BASE] Unregistered BLE device {'did': '1022910617', 'eid': 4106, 'edata': '58', 'pdid': 2480, 'seq': 229}
```

**Comment by AlexxIT** [comment]
Should be fixed in v2.0.2

**Comment by darkkatarsis** [comment]
I confirm, 2.0.2 fixes the problem
